src/aura/AuraAvatarTwoToken.sol

```diff
@@ -587,47 +595,79 @@ contract AuraAvatarTwoToken is BaseAvatar, PausableUpgradeable, AuraAvatarUtils,
         }
     }
 
+    /// @notice A function to process pending BAL and AURA rewards at regular intervals.
+    /// @param _auraPriceInUsd A reference price for AURA in USD (in 8 decimal precision) to compare with TWAP price. Leave as 0 to use only TWAP.
+    function _processRewardsKeeper(uint256 _auraPriceInUsd) internal {
+        uint256 lastClaimTimestampCached = lastClaimTimestamp;
+        uint256 claimFrequencyCached = claimFrequency;
+        if ((block.timestamp - lastClaimTimestampCached) < claimFrequencyCached) {
+            revert TooSoon(block.timestamp, lastClaimTimestampCached, claimFrequencyCached);
+        }
+
+        _processRewards(_auraPriceInUsd);
+    }
+
     /// @notice Claim and process BAL and AURA rewards, selling some of it to USDC and depositing the rest to bauraBAL
     ///         and vlAURA.
     /// @param _auraPriceInUsd A reference price for AURA in USD (in 8 decimal precision) to compare with TWAP price. Leave as 0 to use only TWAP.
     /// @return processed_ An array containing addresses and amounts of harvested tokens (i.e. tokens that have finally
     ///                    been swapped into).
-    function processRewardsInternal(uint256 _auraPriceInUsd) internal returns (TokenAmount[] memory processed_) {
+    function _processRewards(uint256 _auraPriceInUsd) internal returns (TokenAmount[] memory processed_) {
         // 1. Claim BAL and AURA rewards
         (uint256 totalBal, uint256 totalAura) = claimAndRegisterRewards();
 
         // 2. Swap some for USDC and send to owner
         uint256 balForUsdc = (totalBal * sellBpsBalToUsdc) / MAX_BPS;
         uint256 auraForUsdc = (totalAura * sellBpsAuraToUsdc) / MAX_BPS;
 
-        uint256 usdcEarnedFromBal = swapBalForUsdc(balForUsdc);
-        uint256 usdcEarnedFromAura = swapAuraForUsdc(auraForUsdc, _auraPriceInUsd);
+        uint256 usdcEarnedFromBal;
+        if (balForUsdc > 0) {
+            usdcEarnedFromBal = swapBalForUsdc(balForUsdc);
+        }
+        uint256 usdcEarnedFromAura;
+        if (auraForUsdc > 0) {
+            usdcEarnedFromAura = swapAuraForUsdc(auraForUsdc, _auraPriceInUsd);
+        }
 
         uint256 totalUsdcEarned = usdcEarnedFromBal + usdcEarnedFromAura;
 
         address ownerCached = owner();
-        USDC.safeTransfer(ownerCached, totalUsdcEarned);
+        if (totalUsdcEarned > 0) {
+            USDC.safeTransfer(ownerCached, totalUsdcEarned);
+        }
 
         // 3. Deposit remaining BAL to 80BAL-20ETH BPT
-        depositBalToBpt(totalBal - balForUsdc);
+        uint256 balToBpt = totalBal - balForUsdc;
+        if (balToBpt > 0) {
+            depositBalToBpt(totalBal - balForUsdc);
```

Comment:
Why not using `balToBpt ` as the parameter here?

Reply:
Oops, fixed